projects/dsp-ui/src/lib/action/components/sort-button/sort-button.component.spec.ts

```diff
@@ -0,0 +1,166 @@
+import { async, ComponentFixture, TestBed } from '@angular/core/testing';
+
+import { MatIconModule } from '@angular/material/icon';
+import { MatMenuModule } from '@angular/material/menu';
+import { BrowserAnimationsModule } from '@angular/platform-browser/animations';
+import { SortButtonComponent } from './sort-button.component';
+
+import { Component, DebugElement, OnInit, ViewChild, } from '@angular/core';
+import { By } from '@angular/platform-browser';
+import { SortByPipe } from '../../pipes/array-transformation/sort-by.pipe';
+
+/**
+ * Test host component to simulate parent component with a progress bar.
+ */
+@Component({
+    template: `
+    <dsp-sort-button #sortButton [sortProps]="sortProps" [(sortKey)]="sortKey" [position]="position">
+    </dsp-sort-button>
+    <ul class="list">
+            <li *ngFor="let item of list | dspSortBy: sortKey" class="item">
+                <span [class.active]="sortKey === 'firstname'">{{item.firstname}} </span>
+                <span [class.active]="sortKey === 'lastname'">{{item.lastname}} </span>
+                by
+                <span [class.active]="sortKey === 'creator'">{{item.creator}}</span>
+            </li>
+        </ul>
+    `
+})
+class TestHostComponent implements OnInit {
+
+    @ViewChild('sortButton', { static: false }) sortButtonComponent: SortButtonComponent;
+
+    sortProps: any = [{
+        key: 'firstname',
+        label: 'First name'
+    },
+    {
+        key: 'lastname',
+        label: 'Last name'
+    },
+    {
+        key: 'creator',
+        label: 'Creator'
+    }
+    ];
+    sortKey = 'creator';
+    position = 'left';
+
+    list = [{
+        firstname: 'a',
+        lastname: 'z',
+        creator: 'André Franquin'
+
+    },
+    {
+        firstname: 'b',
+        lastname: 'y',
+        creator: 'Walt Disney'
+
+    },
+    {
+        firstname: 'c',
+        lastname: 'x',
+        creator: 'William Shakespeare'
+
+    },
+    {
+        firstname: 'd',
+        lastname: 'w',
+        creator: 'Charles M. Schulz'
+
+    }
+    ];
+
+    constructor() {
+    }
+
+    ngOnInit() { }
+}
+
+describe('SortButtonComponent', () => {
+    let testHostComponent: TestHostComponent;
+    let testHostFixture: ComponentFixture<TestHostComponent>;
+    const listData = [
+        { firstname: 'a', lastname: 'z', creator: 'André Franquin' },
+        { firstname: 'b', lastname: 'y', creator: 'Walt Disney' },
+        { firstname: 'c', lastname: 'x', creator: 'William Shakespeare' },
+        { firstname: 'd', lastname: 'w', creator: 'Charles M. Schulz' },
+    ];
+
+    beforeEach(async(() => {
+        TestBed.configureTestingModule({
+            imports: [
+                MatIconModule,
+                MatMenuModule,
+                BrowserAnimationsModule
+            ],
+            declarations: [
+                SortButtonComponent,
+                TestHostComponent,
+                SortByPipe]
```

Comment:
@mdelez why do you still use the sort-by pipe marked as deprecated? would it make sense to use the new sorting service instead (as explained in the sort-by pipe annotation)?

Reply:
same question here 😉